### PR TITLE
Check for proto drift in the CI

### PR DESCRIPTION
**Pull request by frouioui**
## Description

This pull request adds the `check_make_proto` workflow to the CI. The workflow checks that the `go/vt/proto/*.pb.go` files are in accordance with the `proto/*.proto` files. This is done by running the `make proto` command and diffing `git status`'s output before and after executing the `make proto`.

If there is a diff (a drift), the workflow lists the files that are not in accordance. A commit that exhibits this behavior can be found [here](https://github.com/vitessio/vitess/pull/9644/commits/a96bb3350591db945bdea182e5cb2f1e7159de47), with its failing `check_make_proto` workflow [here](https://github.com/vitessio/vitess/runs/5095360428?check_suite_focus=true).

## Related Issue(s)

- Solves #9588 

## Checklist
- [ ] Should this PR be backported?
- [x] Tests were added or are not required
- [x] Documentation was added or is not required

### CODE REVIEW

File: tools/check_make_proto.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+source build.env
+
+first_output=$(git status --porcelain)
```
Comment (doeg) [inline tools/check_make_proto.sh:5]: Wouldn't an initial `git status` on a fresh checkout always be empty?

**Review by doeg — APPROVED** [review]
I love this! 😻 Thank you so much. 

.......... I will also be ~~stealing~~ borrowing your excellent idea for #9001 >:)

**Review by deepthi — APPROVED** [review]
You may merge after addressing the question from @doeg.
Can you squash-merge, please?

File: tools/check_make_proto.sh (hunk 1 above)
Comment (frouioui) [reply tools/check_make_proto.sh:5]: Yes, technically the first git status should be empty, however, I wanted to make sure that the only possible reason for a drift comes from make proto, no matter how or when this script is being run. So I assumed the script could be run in any situation/environment, and not necessarily on a clean git state :)

**Pull request merged**